src/lib.rs

```diff
@@ -359,6 +374,42 @@ impl Default for VadConfig {
     }
 }
 
+impl VadConfig {
+    pub fn new(
+        positive_speech_threshold: f32,
+        negative_speech_threshold: f32,
+        pre_speech_pad: Duration,
+        redemption_time: Duration,
+        sample_rate: usize,
+        min_speech_time: Duration,
+    ) -> Result<Self> {
+        let config = VadConfig {
+            positive_speech_threshold,
+            negative_speech_threshold,
+            pre_speech_pad,
+            redemption_time,
+            sample_rate,
+            min_speech_time,
+        };
+        match config.validate_config() {
+            Ok(_) => Ok(config),
+            Err(e) => Err(e),
+        }
+    }
+
+    pub fn validate_config(&self) -> Result<()> {
+        #[cfg(feature = "audio_resampler")]
+        return Ok(());
+
+        #[cfg(not(feature = "audio_resampler"))]
+        if ![8000, 16000].contains(&self.sample_rate) {
+            bail!("You can only choose 8000 or 16000Hz sample rate")
```

Comment:
`Invalid sample rate of {}, expected either 8000Hz or 16000Hz` I think is a better message